Features/apps table status change #154064901

callaghanc opened the pull request.

chrisdolendo committed:
WIP

chrisdolendo committed:
[#154064901] - Implement app table status dropdown
- Need to update css of dropdown so it aligns correctly

chrisdolendo committed:
[#154064901] - Init setting up of status change modal

Commit:
Adding css to fix overflow and dropdown button

Commit:
Fixing merge conflicts

Commit:
Table overflow nudge

callaghanc committed:
[#154064901] Makes status modal able to open and close

Commit:
Fixing the ghost nudge

callaghanc committed:
Merge branch 'master' into features/apps-table-status-change-#154064901

Commit:
Updating modal to use button rather than link

callaghanc committed:
[#154064901] Creates a form in status update modal

callaghanc committed:
[#154064901] Adds creation of field update comment to status table

callaghanc committed:
[#154064901] WIP - begins to add alert banner to status update modal

callaghanc committed:
[#154064901] Makes lease up status changes reflected in the apps table

callaghanc committed:
Merge branch 'master' into features/apps-table-status-change-#154064901

callaghanc committed:
[#154064901] Adds alert, loading, and update date for status updates

callaghanc committed:
[#154064901] Updates tests and lease up status style helper

callaghanc commented on app/javascript/components/lease_ups/LeaseUpsTableContainer.js:
```diff
@@ -1,12 +1,90 @@
+      if (response) {
+        // find the application in question in the applications table data
+        // and update its lease up status value and status updated date
+        // in the table
```
@fddayan After we chatted about whether to update data in the table or to pull from Salesforce after the user creates a new field update comment, I started working on the "pull just the updated application" approach we talked about, but it didn't feel quite right. I felt like it was bypassing the normal flow of data btwn Rails and the lease ups table and adding complexity that could be distracting. So I decided in the end to go with the approach of updating the new status and date in the table itself.

fddayan replied:
👍

callaghanc committed:
Merge branch 'master' into features/apps-table-status-change-#154064901

callaghanc committed:
[#154064901] Renames stringUtils

callaghanc committed:
[#154064901] Adds some class names to status modal dropdown

callaghanc commented:
@chrisdolendo I've resolved merge conflicts on this and tests are passing locally, but tests are not passing on Semaphore. I'm over hours for the week, so can you take a look at Semaphore for this tmrw?

callaghanc commented:
@slowbot I messed around a bit with the modal dropdown but haven't yet come across a satisfactory solution to its display issues. I'm done for the week, so can you take a look and see if you can get it to be the right width and position? For its overflow, I actually think it may be better to keep it contained inside the modal, not letting it overflow, and instead have the dropdown menu scroll.

callaghanc committed:
[#154064901] Adds error trigger on status update for testing

Commit:
Adding fixes for full width dropdown and overflow

fddayan committed:
[#154064901] using useFakeTimers in ApplicationNewPage.test.js

chrisdolendo reviewed (approved):
:+1: Great! 🎉

fddayan commented on app/javascript/components/applications/ApplicationDetailsContentCard.js:
```diff
@@ -38,16 +38,16 @@ var generateContent = (dataCollection, field, labelMapper, i) => {
+//       <p>{value}</p>
+//     </div>
+//   )
+// }
```
should we removed this commented code?

callaghanc replied:
not even sure why it was commented out in the first place! we don't use this function anywhere so yes, will remove this commented code

fddayan commented on app/javascript/components/applications/application_form/AlternateContactSection.js:
```diff
@@ -1,4 +1,5 @@
 import React from 'react'
+import _ from 'lodash'
```
bases on grant suggestions we are importing from lodash whatever we are actually using to help performance in our bundle

fddayan commented on app/javascript/components/applications/application_form/HouseholdMembersSection.js:
```diff
@@ -1,4 +1,5 @@
 import React from 'react'
+import _ from 'lodash'
```
same thing here about lodash..not sure if you worked a lot in this components and refactoring this is out of the scope of this PR

callaghanc replied:
yeah what this import is and the one above it is me adding imports where lodash was used but never imported, did so to pass overcommit, which was checking the whole codebase, not just the files i was editing. so even though this story didn't involve these components i added those imports. that was before grant pointed out that we should just import the specific lodash functions we need. i can definitely go back and update these to do that, since i'm bothering to add the imports at all i'll update them to use our new convention

fddayan commented on app/javascript/components/lease_ups/LeaseUpsStatusModalWrapper.js:
```diff
@@ -0,0 +1,61 @@
+        header="Update Status"
+        primary="update"
+        secondary="cancel"
+        isOpen={this.props.isOpen}
```
about `props` I would follow this conventions when using many props: Instead of referencing `this.props.whatever` I would extract the props in the first line in the render method like `const { whatever } = this.props` so you can see what properties are being using in the render method and you do not have to scan the code to see what it's being used. Also, we save a few characters in the JSX, but this is minor improvement :P

callaghanc replied:
nice, i'll update to use that convention, thanks!

fddayan commented on app/javascript/components/lease_ups/LeaseUpsTableContainer.js:
```diff
@@ -1,10 +1,90 @@
+
+  updateState = (path, value) => {
+    this.setState(prevState => {
+      var newState = cloneDeep(prevState)
```
what are we deep cloning the state? whatever we return in it gets merged to the current state. so if you state is  `{a:1, b2}` and you return `{b:3}`. The state becomes `{a1, b:3}`. you do not lose the other keys.
also passing a function to setState is technically used only if you need a value from the `prevState` because react updates the state async. Ping me on slack if this is confusing.

callaghanc replied:
my aim here was to not override nested objects - i'll ping you on slack to discuss further - definitely would be great if we don't have to deep clone!

callaghanc replied:
@fddayan i pushed the new state update functions we worked on

fddayan commented on app/javascript/components/lease_ups/leaseUpsHelpers.js:
```diff
@@ -0,0 +1,24 @@
+ {value: 'Waitlisted', label: 'Waitlisted', style:'is-waitlisted'},
+ {value: 'Disqualified', label: 'Disqualified', style:'is-disqualified'},
+ {value: 'Approved', label: 'Approved', style:'is-approved'},
+ {value: 'Lease Signed', label: 'Lease Signed', style:'is-leased'},
```
👏

fddayan commented on app/javascript/components/molecules/FormModalLoading.js:
```diff
@@ -0,0 +1,12 @@
+import React from 'react'
+import Loading from '../molecules/Loading'
+
+const FormModalLoading = () => {
```
Just adding a note that we might need a loading HOC to handle all the loadings for components. Nothing we have to implement now. Just making a note.

fddayan reviewed (changes requested):
Overall looks good. I added some cleanup changes and a question about why doing deep clone if states

callaghanc committed:
[#154064901] Does some cleanup and slight refactoring

callaghanc committed:
[#154064901] Makes some small corrections to lease up apps table

- Corrects data formatting
- Changes the app number link in table rows to link to app supplementals, not app
- Refactors functions for updating lease up table container state
- Reduces lease up app table's status button dropdown's CSS top value

Commit:
Fixing modal dropdown spacing

callaghanc committed:
[#154064901] Corrects preference rank sort order in the lease up apps table

callaghanc committed:
[#154064901] Updates tests

callaghanc committed:
[#154064901] Updates tests

Commit:
Merge branch 'features/apps-table-status-change-#154064901' of https://github.com/Exygy/sf-dahlia-lap into features/apps-table-status-change-#154064901

Commit:
Fixing table cell spacing for status default coloring

Commit:
Fixing rank order min width

Commit:
Fixing background color on status default table cells

callaghanc committed:
[#154064901] Updates tests

callaghanc committed:
[#154064901] Updates CSS with latest from PL lap-leaseup branch

callaghanc committed:
Merge branch 'master' into features/apps-table-status-change-#154064901
